smt/pkg/smt/smt_state_writer.go

```diff
@@ -0,0 +1,140 @@
+package smt
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/holiman/uint256"
+	"github.com/ledgerwatch/erigon-lib/common"
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/core/state"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/core/types/accounts"
+	"github.com/ledgerwatch/erigon/smt/pkg/utils"
+)
+
+var _ state.StateWriter = (*SMT)(nil)
+
+func (s *SMT) UpdateAccountData(address common.Address, _ *accounts.Account, account *accounts.Account) error {
+	if err := s.SetAccountStorage(address, account); err != nil {
+		return err
+	}
+
+	// TODO: @Stefan-Ethernal How to update:
+	// 1. the account bytecode (we have CodeHash, but not the Code itself, so how to retrieve the actual Code) and
+	// 2. the storage slots (we only have the Root, but not the storage slots themselves)
+
+	return nil
+}
+
+func (s *SMT) UpdateAccountCode(address common.Address, _ uint64, codeHash common.Hash, code []byte) error {
+	if len(code) == 0 {
+		return nil
+	}
+
+	return s.SetContractBytecode(address.Hex(), fmt.Sprintf("0x%s", hex.EncodeToString(code)))
+}
+
+func (s *SMT) DeleteAccount(address common.Address, original *accounts.Account) error {
+	hexAddr := address.Hex()
+
+	keyBalance := utils.KeyEthAddrBalance(hexAddr)
+	keyNonce := utils.KeyEthAddrNonce(hexAddr)
+	keyContractCode := utils.KeyContractCode(hexAddr)
+
+	// TODO: @Stefan-Ethernal Remove contract storage, but how to retrieve the contract storage slots,
```

Comment:
We don't need to remove the storage slots. I doubt this function will be used because an account can't be deleted after Cancun.

Reply:
Ok, in that case, I have left it unimplemented and just returned an error https://github.com/0xPolygonHermez/cdk-erigon/pull/1398/commits/6abc3e1dd25d203de4fc2237a1a2b2d84f1318d7.

IMO, it is better to have it unimplemented, than half implemented.